### PR TITLE
Fix article progress example on android

### PR DESCRIPTION
**Pull request by Latropos**
## Summary

An error is thrown when you measure en element that is not mounted yet. So I create a boolean flag which I set when component is ready to avoid this error.

Probably we also should improve error log.

Current error log:
> `[Reanimated] The view with tag ${viewTag} returned an invalid measurement response.`

New error log:
> `[Reanimated] The view with tag ${viewTag} returned an invalid measurement response. Please make sure the view is currently rendered.`

Both logs are not perfect, since in our example our measured element is a Text component, not a View one, however adding component type to log is not easy.

### CODE REVIEW

File: app/src/examples/ArticleProgressExample.tsx [hunk 1]
```diff
@@ -11,8 +11,16 @@ export default function ArticleProgressExample() {
...
   const textRef = useAnimatedRef<Animated.Text>();
   const scrollHandler = useScrollViewOffset(scrollViewRef);
+  const [mounted, setMounted] = useState(false);
+
+  useEffect(() => {
+    setMounted(true);
+  }, []);
 
   const progressBarAnimatedStyle = useAnimatedStyle(() => {
+    if (!mounted) {
```
Comment (piaskowyk) [inline app/src/examples/ArticleProgressExample.tsx:21]: W shouldn't use state value in worklet, lets change it to shared value.

File: src/reanimated2/NativeMethods.ts [hunk 2]
```diff
@@ -75,7 +75,7 @@ if (isWeb()) {
       return null;
     } else if (measured.x === -1234567) {
       console.warn(
-        `[Reanimated] The view with tag ${viewTag} returned an invalid measurement response.`
+        `[Reanimated] The view with tag ${viewTag} returned an invalid measurement response. Please make sure the view is currently rendered.`
```
Comment (piaskowyk) [inline src/reanimated2/NativeMethods.ts:78]:
```suggestion
        `[Reanimated] The view with tag ${viewTag} returned an invalid measurement response. Please ensure that the view has been rendered.`
```

**Review by piaskowyk — CHANGES_REQUESTED** [review]

File: app/src/examples/ArticleProgressExample.tsx [hunk 3]
```diff
@@ -13,6 +13,10 @@ export default function ArticleProgressExample() {
   const scrollHandler = useScrollViewOffset(scrollViewRef);
 
   const progressBarAnimatedStyle = useAnimatedStyle(() => {
+    if (!textRef.current || !scrollViewRef.current) {
```
Comment (tomekzaw) [inline app/src/examples/ArticleProgressExample.tsx:16]: Looks like we pass React refs to a worklet, do we really want to do it?

Comment (Latropos) [reply app/src/examples/ArticleProgressExample.tsx:16]: Why not? We were already using ref inside worklet:
```javascript
const measuredText = measure(textRef);
```
And this seems to be the best way to check if component is already rendered...

Comment (tomekzaw) [reply app/src/examples/ArticleProgressExample.tsx:16]: @Latropos Turns out it doesn't work as expected, animated refs don't have `.current` property when used on UI runtime. We'll probably need to revert this PR and find another fix. cc @jfedak whom I've complained to about ArticleProgressExample not working on macOS

**Pull request merged**